internal/operands/common-templates/reconcile_test.go

```diff
@@ -285,3 +361,24 @@ func getTestTemplates() []templatev1.Template {
 		},
 	}}
 }
+
+func getCommonTemplatesRestoredMetric() (string, float64) {
+	ch := make(chan prometheus.Metric, 1)
+	CommonTemplatesRestored.Collect(ch)
```

Comment:
Is `.Collect()` guaranteed to only send 1 metric to the channel? Otherwise it would block forever.

The `prometheus.Counter` interface also implements `Metric` interface. Could you use that directly, instead of calling `Collect()`?
```go
CommonTemplatesRestored.Write(dto)
```

Reply:
`Collect` implementation is just `ch <- c.self`, so it is only one.

`Write` works fine, but the Prometheus registry uses `Collect` and not `Write` (which just encapsulates the metric), so I usually prefer to go with `Collect`

Reply:
ok